lib/bamboo/mailer.ex

```diff
@@ -264,10 +280,13 @@ defmodule Bamboo.Mailer do
     build_config(mailer, otp_app)
   end
 
-  def build_config(mailer, otp_app) do
+  def build_config(mailer, otp_app, optional_overrides \\ %{}) do
+    optional_overrides = Enum.into(optional_overrides, %{})
```

Comment:
Is the idea here that people might be passing things in as optional overrides that are in different formats?

Reply:
:thinking: Im trying to recall where my head was when I made this decision.

Im thinking I had a hard time determining if things were passed in as Keyword Lists or Maps, or both. I think I saw this technique in a test, though that scenario wouldnt apply here.

From my current perspective, it feels like defensive programming. I intend to remove it now.

Reply:
Defensively casting the argument as a Map is removed with 8b1abd0f025ffe99a4c9008db2829f05dc0c1b86 :world_map: